modules/apps/transfer/ibc_module.go

```diff
@@ -178,28 +178,38 @@ func (im IBCModule) OnRecvPacket(
 	ack := channeltypes.NewResultAcknowledgement([]byte{byte(1)})
 
 	var data types.FungibleTokenPacketData
+	var ackErr error
 	if err := types.ModuleCdc.UnmarshalJSON(packet.GetData(), &data); err != nil {
-		ack = channeltypes.NewErrorAcknowledgement("cannot unmarshal ICS-20 transfer packet data")
+		ackErr = fmt.Errorf("cannot unmarshal ICS-20 transfer packet data")
+		ack = channeltypes.NewErrorAcknowledgement(ackErr)
+	}
+
+	eventAttributes := []sdk.Attribute{
```

Comment:
Would it make sense to move this declaration/initialization a bit down, just before the check `if ackErr != nil {`? Just for the purpose of code hygiene...